File: 451-sort-characters-by-frequency/sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        hashmap={}
        for c in s:
            if c in hashmap:
               hashmap[c]+=1
            else:
              hashmap[c]=1
        sorted_list=sorted(hashmap.items(),key=lambda x: x[1],reverse=True)

        ans=""
        for chr,freq in sorted_list:
            ans+=chr*freq

        return ans

File: 451-sort-characters-by-frequency/sort-characters-by-frequency.java
class Solution {
    public String frequencySort(String s) {
    
          HashMap<Character,Integer> dict=new HashMap<>();

for(char c: s.toCharArray()){
dict.put(c,dict.getOrDefault(c,0)+1);

}

ArrayList<Map.Entry<Character,Integer>> arr=new ArrayList<>(dict.entrySet());

Collections.sort(arr,(a,b)->b.getValue()-a.getValue());

StringBuilder sb=new StringBuilder();

for(Map.Entry <Character,Integer> entry : arr){
    int freq=entry.getValue();
    char ch=entry.getKey();
    for(int i=0;i<freq;i++){
        sb.append(ch);
    }
}
return sb.toString();
    }
    
}